PowerEditor/src/MISC/Common/FileInterface.h

```diff
@@ -51,6 +51,10 @@ class Win32_IO_File final
 	unsigned long read(void *rbuf, unsigned long buf_size);
 	bool write(const void *wbuf, unsigned long buf_size);
 
+	bool writeStr(const char* str) {
+		return write(str, static_cast<unsigned long>(strlen(str)));
+	};
+

```

Comment:
@donho ,

That method is a good addition but IMO it is good to also add another `write()` that accepts `const std::string& str` as parameter directly and use `str.length()` as size inside.
Using `strlen()` on `string.c_str()` is not optimal because `strlen()` will cycle through all string chars to find the terminating `\0` and find the length and that is just wasting time since `std::string` method `length()` directly has the length. The bigger the string is the better this optimization will be.